### PR TITLE
libuv: uv_fsevents: fall back to kqueue on older mac systems

### PR DESCRIPTION
**Pull request by kencu**
see: https://github.com/libuv/libuv/pull/2290
update Tiger patches

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

10.4, 10.5, 10.6, 10.14
###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/libuv/Portfile [hunk 1]
```diff
@@ -61,17 +62,21 @@ platform darwin {
...
+        patchfiles-append patch-kqueue-on-older-mac-systems.diff
 
         # deprecate the devel port, if installed
         # can be removed after 20201010
         subport libuv-devel {
             PortGroup   obsolete 1.0
             replaced_by libuv
             version     1.28.0
-            revision    0
+            revision    1
```
Comment (michaelld) [inline devel/libuv/Portfile]: I don't think this tweak is necessary.

File: devel/libuv/files/patch-libuv-1-23-2-tiger.diff [hunk 2]
```diff
@@ -1,56 +1,24 @@
...
+diff -u src/unix/fs.c.orig src/unix/fs.c
+--- src/unix/fs.c.orig   2019-05-04 14:35:17.000000000 -0700
++++ src/unix/fs.c        2019-05-04 14:40:00.000000000 -0700
+@@ -729,7 +729,7 @@
  
      return -1;
    }
 -#elif defined(__APPLE__)           || \
-+#elif defined(__APPLE__) && MAC_OS_X_VERSION_MAX_ALLOWED >= 1050  || \
++#elif defined(__APPLE__) && MAC_OS_X_VERSION_MAX_ALLOWED >= 1050           || \
```
Comment (michaelld) [inline devel/libuv/files/patch-libuv-1-23-2-tiger.diff]: I think this whitespace isn't necessary ... yes?

**Comment by michaelld** [comment]
If I understand here, this PR allows the current libuv release to build on any OSX supported by MacPorts ... yes? Does it work on both PPC and Intel (guessing so, but thought it worth asking)? If so & so, then if not done already let's tweak the Portfile to go back to release & devel for all OSX. Can we patch all OSX versions the same now too, or will some patches not work with the OSX they are not intended for? I'll try this (or whatever state the PR is in) on Monday at work with my old OSX boot bots.

File: devel/libuv/files/patch-libuv-1-23-2-tiger.diff (hunk 2 above)
Comment (michaelld) [reply devel/libuv/files/patch-libuv-1-23-2-tiger.diff]: Also probably want this clause in () because they go together outside the ||s.

**Comment by kencu** [comment]
builds on 10.4 PPC and up. Once that libuv PR is integrated, our patches will be simpler, and we can consider rolling the 10.5/10.6 part into the main part, indeed.

The Tiger patches are somewhat more fragile, and I can't promise to keep them up to date as frequently as libuv issues releases :>.

FYI, three tests do not pass on the 10.6, four or fix 10.5, and about 10 tests fail on 10.4. They are the more esoteric features (watching directories, that kqueue can't easily do, ipv6 stuff). Some things I might be able to fix someday -- some, maybe not.

But the only port that really exercises libuv to the limit is nodejs, I think. The other ports that use it, like cmake, are using more thoroughly trodden paths and show no hiccups.

File: devel/libuv/files/patch-libuv-1-23-2-tiger.diff (hunk 2 above)
Comment (kencu) [reply devel/libuv/files/patch-libuv-1-23-2-tiger.diff]: done

File: devel/libuv/Portfile (hunk 1 above)
Comment (kencu) [reply devel/libuv/Portfile]: done

**Comment by kencu** [comment]
I'll take all that as an approval, Michael -- I know you're hopping busy these days!

**Comment by michaelld** [comment]
Yeah I never got around to testing this change; busy-ness abounds! I'll get there sooner or later with my older OSX boot drives as I update them. Thx for getting this together!

**Comment by michaelld** [comment]
This is where I really value the 24 hour or 48 hour timeout period! I had plenty of opportunities, but this never got high enough on my queue ... and it's something we want in MP so IMHO better to get it merged & "in the wild" for testing by MP users!

**Pull request merged**